src/app/settings/action-triggers/components/edit-action-triggers/edit-action-triggers.component.ts

```diff
@@ -0,0 +1,174 @@
+      } else {
+        this.actionForm.reset()
+        this.selectedAction = null
+      }
+    }
+  }
+
+  private patchValue(actionValue: ActionTrigger<KeyboardAndSelectionTriggerData>): void {
+    this.actionForm.patchValue(actionValue)
+    this.addArrValues(actionValue.data.keys, this.attributesArray)
+  }
+
+  private clearFormArray(formArray: UntypedFormArray): void {
+    while (formArray.length !== 0) {
+      formArray.removeAt(0)
+    }
+  }
+
+  public onKeyDown(event: KeyboardEvent): void {
+    event.preventDefault()
+    const newKeyCode: string = SHORTCUT_KEYS_MAPPINGS[event.code] ? SHORTCUT_KEYS_MAPPINGS[event.code] : event.code
+    if (this.keyPress) {
+      const currentKeys: string[] = this.actionForm?.get('data')?.get('keys')?.value ? this.actionForm?.get('data')?.get('keys')?.value : []
+      if (currentKeys.findIndex(keyCode => keyCode === newKeyCode) === -1) {
+        currentKeys.push(newKeyCode)
+        this.createKey(newKeyCode, this.attributesArray)
+      }
+    } else {
+      this.clearFormArray(this.attributesArray)
+      this.createKey(newKeyCode, this.attributesArray)
+      this.keyPress = true
+    }
+  }
+
+  private get attributesArray(): UntypedFormArray {
+    return this.actionForm?.get('data')?.get('keys') as UntypedFormArray
+  }
+
+  private addArrValues(items: string[], formArray: UntypedFormArray): void {
+    for (const item of items) {
+      this.createKey(item, formArray)
+    }
+  }
+
+  private createKey(item: string, formArray: UntypedFormArray): void {
+    const control = new UntypedFormControl(item)
+    formArray.push(control)
+  }
+
+  public selectNewActionTrigger(action: Tv2PartAction): void {
+    this.actionForm.patchValue({ actionId: action.id })
+    this.selectedAction = action
+  }
+
+  public onKeyUp(): void {
+    this.keyPress = false
+  }
+
+  public get isUpdateAction(): boolean {
+    return !!this.selectedActionTrigger && !!this.selectedActionTrigger.id
+  }
+
+  public submit(): void {
+    this.submitted = true
+    if (this.actionForm.invalid) {
+      this.actionForm.markAllAsTouched()
+      return
+    }
+    this.submitting = true
+    const actionTriggerValue = this.prepareSendData(this.actionForm.value)
+    if (this.isUpdateAction) {
+      this.updateActionTrigger({ ...actionTriggerValue, id: this.selectedActionTrigger?.id as string })
+    } else {
+      this.createActionTrigger(actionTriggerValue)
+    }
+  }
+
+  private prepareSendData(formData: ActionTriggerUIForm): CreateActionTrigger<KeyboardTriggerData> {
+    const keysArray: string[] = formData.data?.keys
+    const resultData: CreateActionTrigger<KeyboardTriggerData> = {
+      actionId: formData.actionId,
+      data: { keys: keysArray, label: formData.data.label ? formData.data.label : (this.selectedAction as Tv2PartAction).name },
+    }
+    if (this.selectedAction?.argument) {
+      resultData.data.actionArguments = +formData.data.actionArguments
+    }
```

Comment:
Just a note, that this should only be applied if the action argument type is an integer.

Reply:
Yes I saw it and fix it last week, new logic is:
if (this.selectedAction?.argument) {
      resultData.data.actionArguments = formData?.data?.actionArguments && this.selectedAction?.argument.type === ArgumentType.NUMBER ? +formData?.data?.actionArguments : formData.data.actionArguments
    } else {
      delete resultData.data.actionArguments
    }`